src/ntcore_c.cpp

```diff
@@ -436,3 +436,174 @@ void NT_DisposeRpcCallInfo(NT_RpcCallInfo *call_info) {
+	value->type = NT_Type::NT_BOOLEAN;
+
+	*set_value_ret = NT_SetEntryValue(name, name_len, value);
+	return value;
+}
+
+struct NT_Value *NT_SetNTValueString(const char *name, size_t name_len, struct NT_String v_string, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_InitValue(value);
+	value->data.v_string = v_string;
+	value->type = NT_Type::NT_STRING;
+
+	*set_value_ret = NT_SetEntryValue(name, name_len, value);
+	return value;
+}
+
+struct NT_Value *NT_SetNTValueRaw(const char *name, size_t name_len, struct NT_String v_raw, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	NT_InitValue(value);
+	value->data.v_raw = v_raw;
+	value->type = NT_Type::NT_RAW;
+
+	*set_value_ret = NT_SetEntryValue(name, name_len, value);
+	return value;
+}
+
+
+
+struct NT_Value *NT_SetNTValueBooleanArr(const char *name, size_t name_len, int *arr, size_t size, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	value->type = NT_Type::NT_DOUBLE_ARRAY;
+	value->data.arr_boolean.arr = arr;
+	value->data.arr_boolean.size = size;
+	return value;
+}
+
+struct NT_Value *NT_CreateNTValueDoubleArr(const char *name, size_t name_len, double *arr, size_t size, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	value->type = NT_Type::NT_DOUBLE_ARRAY;
+	value->data.arr_double.arr = arr;
+	value->data.arr_double.size = size;
+	return value;
+}
+
+struct NT_Value *NT_SetNTValueNTStringArr(const char *name, size_t name_len, struct NT_String *arr, size_t size, int force, int *set_value_ret)
+{
+	NT_Value *value = static_cast<NT_Value*>
+		(std::malloc(sizeof(struct NT_Value)));
+	value->type = NT_Type::NT_DOUBLE_ARRAY;
+	value->data.arr_string.arr = arr;
+	value->data.arr_string.size = size;
+	return value;
+}
+
+void NT_GetDataFromNTValue(struct NT_Value *value,
+	void *data, void **data_arr, size_t *size, enum NT_Type *type,
+	unsigned long long *last_change)
+{
+	*type = value->type;
+	*last_change = value->last_change;
+	switch (value->type)
+	{
+	case NT_Type::NT_BOOLEAN:
+		data = &value->data.v_boolean;
+		*size = 0;
+		break;
+	case NT_Type::NT_DOUBLE:
+		data = &value->data.v_double;
+		*size = 0;
+		break;
+	case NT_Type::NT_STRING:
+		data = &value->data.v_string;
+		*size = 0;
+		break;
+	case NT_Type::NT_RAW:
+		data = &value->data.v_raw;
+		*size = 0;
+		break;
+	case NT_Type::NT_BOOLEAN_ARRAY:
+		*data_arr = value->data.arr_boolean.arr;
+		*size = value->data.arr_boolean.size;
+		break;
+	case NT_Type::NT_DOUBLE_ARRAY:
+		*data_arr = value->data.arr_double.arr;
+		*size = value->data.arr_double.size;
+		break;
+	case NT_Type::NT_STRING_ARRAY:
+		*data_arr = value->data.arr_string.arr;
+		*size = value->data.arr_string.size;
+		break;
+	default:
+		data = nullptr;
```

Comment:
A couple of things. First, we don't want to change data here. What we will want to do is set *data_arr to be NULL.